crates/block-producer/src/state_view/mod.rs

```diff
@@ -82,17 +82,25 @@ where
         // after getting the transaction inputs.
         ensure_in_flight_constraints(
             candidate_tx,
-            &*self.accounts_in_flight.read().await,
             &*self.nullifiers_in_flight.read().await,
             &*self.notes_in_flight.read().await,
             &[],
         )?;
+        self.accounts_in_flight.read().await.verify_update(
+            candidate_tx.account_id(),
+            candidate_tx.account_update().init_state_hash(),
+        )?;
```

Comment:
This check is now external to `ensure_in_flight_constraints` which I'm not entirely happy with.

Its easy to move it back inside but then the second check that performs the update would do extra work (verifying twice). 

Options:
1. Leave as is
2. Move check back inside, checking twice is not the end of the world
3. Remove the check from `verify_and_add` (not a fan of this)
4. Some more type safe API which only lets you update after verification is confirmed. I did toy around with this, but its substantially more code.

Wasn't quite happy with any of these. Open to suggestions.

Reply:
I think I'd prefer the second option (do the check twice).

We could also change `InflightAccountStates::verify_and_add()` into just `InflightAccountStates::add()` - but I think the check is light enough where doing it twice shouldn't have a meaningful impact on things.